database/database.go

```diff
@@ -4,41 +4,37 @@ import (
 	"context"
 	"time"
 
-	"github.com/dezh-tech/immortal/types"
 	"go.mongodb.org/mongo-driver/bson"
 	"go.mongodb.org/mongo-driver/mongo"
 	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
-var KindToCollectionName = map[types.Kind]string{
-	types.KindTextNote:        "text_notes",
-	types.KindReaction:        "reactions",
-	types.KindProfileMetadata: "profile_metadatas",
-}
-
 type Database struct {
 	DBName       string
 	QueryTimeout time.Duration
 	Client       *mongo.Client
 }
 
-func New(cfg Config) (*Database, error) {
+func Connect(cfg Config) (*Database, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(cfg.ConnectionTimeout)*time.Millisecond)
+	defer cancel()
+
 	serverAPI := options.ServerAPI(options.ServerAPIVersion1)
 	opts := options.Client().ApplyURI(cfg.URI).
 		SetServerAPIOptions(serverAPI).
-		SetConnectTimeout(time.Duration(cfg.ConnectionTimeout) * time.Millisecond). // Convert to time.Duration
+		SetConnectTimeout(time.Duration(cfg.ConnectionTimeout) * time.Millisecond).
 		SetBSONOptions(&options.BSONOptions{
 			UseJSONStructTags: true,
 			NilSliceAsEmpty:   true,
 		})
 
-	client, err := mongo.Connect(context.TODO(), opts)
+	client, err := mongo.Connect(ctx, opts)
 	if err != nil {
 		return nil, err
 	}
 
 	var result bson.M
-	if err := client.Database("admin").RunCommand(context.TODO(), bson.D{{Key: "ping", Value: 1}}).
+	if err := client.Database("admin").RunCommand(ctx, bson.D{{Key: "ping", Value: 1}}).
```

Comment:
use `queryTimeout`

Reply:
then we need to create 2 contexts. is this logical?